### ISSUE TITLE
Docusaurus Migration

### ISSUE THREAD

**Issue by amydevs** [MatrixAI/Polykey-Network-Dashboard#1]
### Specification

The current implementation of the dashboard is custom SSR code attached to Fastify that serves up the React page and bundle. This needs to be changed to use Docusaurus instead.

### Docusaurus Application

Docusaurus will allow us to take advantage of SSG, so that react pages are prerendered. 

As such, the structure from Polykey.com must be copied over in order to use Cloudflare workers to serve up the webpages.
 
### API Endpoint

The backend for the API endpoints will continue to use Fastify. This means that they will continue to be deployed on EC2 instances, but with the serving of the React SPA removed.

### Tasks

1. Migrate SPA to use Docusaurus
2. Remove SPA from Fastify Code

**Comment by CMCDragonkai** [comment]
Based on changes in polykey.com docusaurus site I think there's a few files in this repo that is not necessary.

1. `babel.config.js`
2. The `URL` `.env.example` does that really make sense here?

**Comment by CMCDragonkai** [comment]
I have a feeling that:

```
# URL the worker is deployed to
URL=https://testnet.polykey.com
```

Can be hardcoded. But instead you just switch between environments in the CI.

For example I have `name = "polykey-com-dev"` in `wrangler.toml` in Polykey.com.

Interestingly in this case, the `testnet.polykey.com` __is__ the "dev" version of the real thing being `mainnet.polykey.com`. I don't think we need another "dev" version do we? Unless you want there to be a dev version just to separate staging from master?

**Comment by amydevs** [comment]
the URL is necessary because, otherwise we will not know which hostname to send API requests when starting locally

**Comment by amydevs** [comment]
it can't be hardcoded, because i need to point it to `http://localhost:3000` locally

**Comment by CMCDragonkai** [comment]
So the URL is necessary for the dashboard to know what API to call into? Ok right so it's because it's a frontend and backend separated project.

So that means when working on this, you have to have both a frontend and backend together, but it can be difficult to orchestrate such a working environment.

What you want to do here is one of 2 things:

1. Is the frontend a dependency of the backend?
2. Is the backend a dependency of the frontend?
3. Are they mutually separate and therefore can work independently?

I think in this case the frontend depends on the backend existing.

One way to deal with this is a submodule dependency.

However in this case, I don't think it can work because the backend is proprietary.

So in this case we need to "orchestrate" it from the outside instead. Lacking this, you need to describe in the README.md how to run the backend. And how that sets the URL.

Note that the frontend URL and the backend URL doesn't have to be the same thing here, even if it is in production.

So I would never use `URL` for this. I would always use `POLYKEY_NETWORK_STATUS_HOST` and `POLYKEY_NETWORK_STATUS_PORT` and join them together.

**Comment by CMCDragonkai** [comment]
Has this been done @amydevs?

**Comment by CMCDragonkai** [comment]
Alot of the README.md and commentary inside the `worker.ts` hasn't been upgraded. I suggest you to also do some chore commits to clean up this repo.

**Comment by CMCDragonkai** [comment]
Get rid of all the useless files too. Like `babel.config.js`. I suggest you to standardise the documentation on README.md with polykey.com and matrix.ai websites.

**Comment by CMCDragonkai** [comment]
What was the reason to have `postcss.config.js`? Was this coming from Polykey-Enteprise? I don't think docusaurus makes use of it.

**Comment by CMCDragonkai** [comment]
I noticed `polykey.com` cargo culted some of these files that are not relevant. That repo should also be cleaned up along with this.

**Comment by CMCDragonkai** [comment]
I'm trying to replicate what you're doing regarding chart.js for Tetra a dashboard experiment. But the code inside `src/components/ResourceChart.tsx` is a bit hard to follow. What's with the dynamic imports of the charts? Why not just import them at the top? We should review the standardisation of the frontend.

**Comment by amydevs** [comment]
the dynamic imports are so that docusaurus doesn't try rendering importing the charting libraries during SSG, as this will cause the SSG builder to complain that something has tried to access `window` (and it doesn't exist within the Node environemnet).

**Comment by amydevs** [comment]
> What was the reason to have `postcss.config.js`? Was this coming from Polykey-Enteprise? I don't think docusaurus makes use of it.

postcss is used for tailwind

**Comment by CMCDragonkai** [comment]
> the dynamic imports are so that docusaurus doesn't try rendering importing the charting libraries during SSG, as this will cause the SSG builder to complain that something has tried to access `window` (and it doesn't exist within the Node environemnet).

Can you make this more obvious? Maybe by grouping it into a utility and documenting it?

**Comment by CMCDragonkai** [comment]
> > What was the reason to have `postcss.config.js`? Was this coming from Polykey-Enteprise? I don't think docusaurus makes use of it.
> 
> postcss is used for tailwind

I don't see that being used by docusaurus?

**Comment by amydevs** [comment]
i am using tailwind utility classes in my dashboard

**Comment by CMCDragonkai** [comment]
Last thing is changing the environment variables.

**Comment by CMCDragonkai** [comment]
These 2:

* `POLYKEY_NETWORK_STATUS_HOST`
* `POLYKEY_NETWORK_STATUS_PORT`

May be all that is necessary if you are able to use protocoless URL. If you cannot, then add another one:

```
POLYKEY_NETWORK_STATUS_PROT=http/https
```

**Comment by CMCDragonkai** [comment]
You'll need to update the cloudflare too to close this issue.

**Comment by CMCDragonkai** [comment]
Ok those are actually build time variables. I've fixed this now with 7746bc2ac5b4661d465f7701806573118fd6c2c6
